elixir/apps/api/lib/api/gateway/channel.ex

```diff
@@ -150,7 +151,9 @@ defmodule API.Gateway.Channel do
             flow_id: flow_id,
             resource: Views.Resource.render(resource),
             expires_at: DateTime.to_unix(authorization_expires_at, :second),
-            payload: payload
+            payload: payload,
+            client_ipv4: client.ipv4,
+            client_ipv6: client.ipv6
```

Comment:
I decided to flatten these two fields into the message because that is all we need on the client. Sending everything would also work but only sending partial structs is annoying because it means I need to create two almost identical data structures for deserialisation.

@jamilbk @AndrewDryga Let me know if you are happy with this or whether you'd rather send everything at once.

I want to note that in the future we'd like to de-couple the "allow-access" fields that are currently embedded in every "request connection" message so the current concept of a "Client"-view might go away because the different fields are used for different things:

- The IPs are only necessary for allowing access
- The public key is only necessary when setting up a new connection
- We ignore the keep-alive field already
- The pre-shared key is also only necessary for setting up a new connection

Given that, I think adding only the fields that we need here makes more sense.

Reply:
Flattening and using individual fields makes the structure less opinionated, which might future proof it more I suppose.